compaction/compaction_manager.cc

```diff
@@ -697,7 +697,9 @@ compaction_manager::compaction_reenabler::~compaction_reenabler() {
     if (_table && --_compaction_state.compaction_disabled_counter == 0 && !_compaction_state.gate.is_closed()) {
         cmlog.debug("Reenabling compaction for {}", *_table);
         try {
-            _cm.submit(*_table);
+            // We can safely call get_candidates here since
+            // compaction is known to be disabled.
+            _cm.submit(*_table, _cm.get_candidates(*_table));
```

Comment:
Why not do this inside `submit()` improving it for all callers?

Reply:
I'm not sure calling get_candidates in all paths calling submit is safe.
Here, we know it's safe in the compaction reenabler since we're sure compaction is disabled.